app/src/main/assets/GimelGimel/VectorLayer.js

```diff
@@ -57,15 +61,18 @@ GG.Layers.VectorLayer.prototype.addMarker = function (id, location, symbol) {
  */
 GG.Layers.VectorLayer.prototype.updateMarker = function (id, location, symbol) {
     GG.Utils.assertIdExists(id, this._entities);
-
     var marker = this._entities[id];
-    if (location) {
-        marker.position = Cesium.Cartesian3.fromDegrees(location.longitude, location.latitude, location.altitude);
-    }
-    if (symbol) {
-        clearSymbol(marker);
-        setMarkerSymbology(marker, symbol);
-    }
+
+    addAltitudeToLocation(location).then(function (newLocation) {
```

Comment:
Rename newLocation to explain that it now contains height and not just a whole newLocation object.
Perhaps clippedToGroundLocation ?